packages/resolve-es/src/index.js

```diff
@@ -1,20 +1,35 @@
-function createDriver({ driver }) {
-    let onEventSavedCallbacks = [];
+import 'regenerator-runtime/runtime';
+import { Readable, Writable } from 'stream';
 
-    return {
-        saveEvent: event =>
-            driver.saveEvent(event).then(() => onEventSavedCallbacks.forEach(cb => cb(event))),
-
-        loadEventsByTypes: driver.loadEventsByTypes,
-
-        loadEventsByAggregateId: driver.loadEventsByAggregateId,
-
-        onEventSaved: (callback) => {
-            onEventSavedCallbacks.push(callback);
-            return () =>
-                (onEventSavedCallbacks = onEventSavedCallbacks.filter(item => item !== callback));
-        }
-    };
-}
-
-export default createDriver;
+export default ({ storage, bus }) => ({
+    getStreamByEventTypes(eventTypes) {
+        const eventStream = Readable({ objectMode: true, read() {} });
+        storage
+            .loadEventsByTypes(eventTypes, event => eventStream.push(event))
```

Comment:
What is the point of mixing *promise* and *async/await*.
*async/await* should be used with here.